sycl/CMakeLists.txt

```diff
@@ -139,10 +139,12 @@ install(DIRECTORY "${sycl_inc_dir}/." DESTINATION "${LLVM_INST_INC_DIRECTORY}" C
 
 # SYCL runtime library
 add_subdirectory(source)
+add_subdirectory(plugins)
 
 # SYCL toolchain builds all components: compiler, libraries, headers, etc.
 add_custom_target( sycl-toolchain
   DEPENDS sycl
+          pi_opencl
```

Comment:
This could be moved into the plugins' CMake, e.g. in ` sycl/plugins/opencl/CMakeLists.txt`:

```cmake
add_dependencies(sycl-toolchain pi_opencl)
```

Adding a plugin would then only rquire `add_subdirectory` call in ` sycl/plugins/CMakeLists.txt`. As plugin authors would copy the CMakeLists.txt of existing plugins it becomes harder to accidentally forget to set up this dependency.

(Also, according to the [CMake docs](https://cmake.org/cmake/help/v3.16/command/add_custom_target.html), to add targets as dependencies to a custom target `add_dependencies` should be used - but that belongs into a different MR).

Reply:
Though we would have to move the `add_subdirectory(plugins)` below the definition of the custom `sycl-toolchain` target.